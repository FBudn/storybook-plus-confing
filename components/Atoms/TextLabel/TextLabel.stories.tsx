import React from "react";
import { View } from "react-native";
import TextLabel from "./TextLabel";

const TextLabelMeta = {
  title: "Atoms/TextLabel",
  component: TextLabel,
  args: {
    children: "TextLabel",
    testId: "test-textLabel-id",
    color: "red",
  },
  decorators: [
    (Story: any) => (
      <View
        style={{
          alignItems: "center",
          justifyContent: "center",
          flex: 1,
          backgroundColor: "white",
        }}
      >
        <Story />
      </View>
    ),
  ],
};

export default TextLabelMeta;

export const Basic = {};
