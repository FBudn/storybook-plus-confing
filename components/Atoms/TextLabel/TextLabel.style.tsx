import { FlexAlignType, StyleSheet } from "react-native";

const TextLabelStyles = (
  color?: string | undefined,

  fontSize?: number | undefined,

  alignItems?: FlexAlignType | undefined,

  textAlign?: any,

  width?: string | number
) =>
  StyleSheet.create({
    text: {
      color: `${color}`,
      fontSize,
      width: "100%",
      textAlign,
    },
    container: {
      width: "100%",
      alignItems,
    },
  });

export default TextLabelStyles;
