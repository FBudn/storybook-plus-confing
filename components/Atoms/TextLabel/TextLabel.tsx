import React from "react";
import { StyleSheet, Text, View } from "react-native";

export interface TextLabelProps {
  children: React.ReactNode;
  color: string | undefined;
  fontSize: number | undefined;
  testId: string | undefined;
}

const TextLabel: React.FC<TextLabelProps> = ({
  children,
  testId,
  color,
  fontSize,
}) => {
  return (
    <View testID={testId}>
      <Text style={styles(color, fontSize).text}>{children}</Text>
    </View>
  );
};

const styles = (
  color?: string | undefined,

  fontSize?: number | undefined
) =>
  StyleSheet.create({
    text: {
      color: `${color}`,
      fontSize,
    },
  });

export default TextLabel;

/*
import React from "react";
import { StyleSheet, Text, View } from "react-native";

export interface TextLabelProps {
  children: React.ReactNode;
  color: string | undefined;
  align:
    | "flex-start"
    | "flex-end"
    | "center"
    | "stretch"
    | "space-between"
    | "space-around"
    | undefined;
  justify:
    | "flex-start"
    | "flex-end"
    | "center"
    | "space-between"
    | "space-around"
    | "space-evenly"
    | undefined;
  fontWeight:
    | "bold"
    | "normal"
    | "100"
    | "200"
    | "300"
    | "400"
    | "500"
    | "600"
    | "700"
    | "800"
    | "900"
    | undefined;
  fontSize: number | undefined;
  testId: string | undefined;
  width?: string | undefined;
  margin?: string | undefined;
}

const TextLabel: React.FC<TextLabelProps> = ({
  children,
  testId,
  color,
  align,
  justify,
  fontWeight,
  fontSize,
  width,
  margin,
}) => {
  return (
    <View
      testID={testId}
      style={
        styles(color, align, justify, fontWeight, fontSize, width, margin)
          .container
      }
    >
      <Text>{children}</Text>
    </View>
  );
};

const styles = (
  color?: string | undefined,
  align?:
    | "flex-start"
    | "flex-end"
    | "center"
    | "stretch"
    | "space-between"
    | "space-around"
    | undefined,
  justify?:
    | "flex-start"
    | "flex-end"
    | "center"
    | "space-between"
    | "space-around"
    | "space-evenly"
    | undefined,
  fontWeight?:
    | "bold"
    | "normal"
    | "100"
    | "200"
    | "300"
    | "400"
    | "500"
    | "600"
    | "700"
    | "800"
    | "900"
    | undefined,
  fontSize?: number | undefined,
  width?: string | undefined,
  margin?: string | undefined
) =>
  StyleSheet.create({
    container: {
      width: `${width}`,
      margin: `${margin}`,
      alignContent: align,
      justifyContent: justify,
      color: `${color}`,
      fontWeight,
      fontSize,
    },
  });

export default TextLabel;

*/
