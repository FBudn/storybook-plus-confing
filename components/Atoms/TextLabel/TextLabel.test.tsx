import React from "react";
import { render, screen } from "@testing-library/react-native";
import TestRenderer from "react-test-renderer";
import TextLabel, { TextLabelProps } from "./TextLabel";

const props: TextLabelProps = {
  color: "black",
  fontSize: 10,
  testId: `test-textLabel-id`,
  children: `TextLabel Test`,
};

test(`Should render TextLabel component`, () => {
  render(<TextLabel {...props} />);
  const TextLabelElement = screen.getByTestId(`test-textLabel-id`);
  expect(TextLabelElement).toBeOnTheScreen();
});

test(`Should match snapshot`, () => {
  const TextLabelElement = TestRenderer.create(
    <TextLabel {...props} />
  ).toJSON();
  expect(TextLabelElement).toMatchSnapshot();
});

/* test(`Should render and have styles:`, () => {
  render(<TextLabel {...props} />);
  const TextLabelElement = screen.getByTestId(`test-textLabel-id`);
  expect(TextLabelElement).toBeOnTheScreen();

  expect(TextLabelElement).toHaveStyle({
    display: "flex",
    border: "none",
  });
}); */

test(`Should render with text`, () => {
  render(<TextLabel {...props} />);
  const TextLabelElement = screen.getByTestId(`test-textLabel-id`);
  expect(TextLabelElement).toBeOnTheScreen();
  expect(TextLabelElement).toHaveTextContent(`TextLabel Test`);
});
