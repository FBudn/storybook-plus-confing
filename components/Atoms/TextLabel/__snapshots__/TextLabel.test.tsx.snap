// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match snapshot 1`] = `
<View
  style={
    {
      "alignItems": undefined,
      "width": "100%",
    }
  }
  testID="test-textLabel-id"
>
  <Text
    style={
      {
        "color": "black",
        "fontSize": 10,
        "textAlign": undefined,
        "width": "100%",
      }
    }
  >
    TextLabel Test
  </Text>
</View>
`;
